src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -234,7 +229,7 @@ public async ValueTask DisposePendingAsyncDisposables()
                 ExceptionDispatchInfo.Capture(exception).Throw();
             }
 
-            PendingAsyncDisposables.Clear();
+            CompletedAsyncDisposables.Clear();
```

Comment:
I assume this doesn't really need to be cleared, since the root serialization call will throw and we will not be able to resume. In any case, clearing here is good.

Reply:
In this case it actually does, this method is meant to clear completed enumerators in the happy path. Serialization could resume after that so we need to avoid instances being disposed more than once.